src/main/frontend/src/components/shopPage/ActiveFilters.js

```diff
@@ -0,0 +1,52 @@
+import styles from "./ActiveFilters.module.css";
+
+function ActiveFilters({selectedCategories, selectedSubcategories, priceRange, categoriesList, onRemoveCategoryClick, onRemoveSubcategoryClick, onRemovePriceFilterClick, onClearAllClick}) {
+
+    const findCategoryNameById = (categoryId) => {
+        const category = categoriesList.find((category) => category.id == categoryId);
+        if (category)
+            return category.name;
+    }
+
+    const categoryFilters = selectedCategories.filter((category) => category != 0);
+
+    return (
+        <div>
+            {(categoryFilters.length > 0 || selectedSubcategories.length > 0 || priceRange[0] > 0 || priceRange[1] < 200) &&
+                (<div className={styles.allFiltersContainer}>
+                    {categoryFilters.length > 0 && 
+                        <div className={styles.filterContainer}>
+                            <p className={styles.filterTitle}>Categories</p>
+                            {categoryFilters.map((category) => 
+                                (<button className={styles.filter} onClick={() => onRemoveCategoryClick(category)}>
+                                    {findCategoryNameById(category)} <span className={styles.removeIcon}>x</span> 
+                                </button>)
+                            )}
+                        </div>
```

Comment:
since this is same pattern for categories and subCategories, we can extract it to function like:
```suggestion
const renderCategoryActiveFilter = (categories, onRemove) => {
return (
                        <div className={styles.filterContainer}>
                            <p className={styles.filterTitle}>Categories</p>
                            {categories.map((category) => 
                                (<button className={styles.filter} onClick={() => onRemove(category)}>
                                    {findCategoryNameById(category)} <span className={styles.removeIcon}>x</span> 
                                </button>)
                            )}
                        </div>
                        )}
```
and then here can be called 
renderCategoryActiveFilter(categoryFilters, onRemoveCategoryClick);
renderCategoryActiveFilter(selectedSubcategories, onRemoveSubcategoryClick);